modules/home-manager/firefox.nix

```
@@ -0,0 +1,67 @@
+{ catppuccinLib }:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (config.catppuccin) sources;
+
+  mkFirefoxModule =
+    name:
+    let
+      cfg = config.catppuccin.${name};
+
+      # ID for the chosed theme's addon
+      addonId = sources.firefox.addonIds.${cfg.flavor}.${cfg.accent};
+      upperName = catppuccinLib.mkUpper name;
+    in
+    {
+      options.catppuccin.${name} =
+        catppuccinLib.mkCatppuccinOption {
+          inherit name;
+          accentSupport = true;
+        }
+        // {
+          finalPackage = lib.mkOption {
+            type = lib.types.package;
+            readOnly = true;
+            description = "Resulting ${upperName} package";
+          };
+          profiles = lib.mkOption {
+            type = with lib.types; listOf str;
+            description = "${upperName} profiles to install the Catppuccin theme addon on";
+            default = [ "default" ];
+            example = lib.literalExpression ''[ "default" "default-release" "work" ]'';
+          };
```

I don't really think there should be a default value in here.

```suggestion
          profiles = lib.mkOption {
            type = with lib.types; listOf str;
            description = "${upperName} profiles to install the Catppuccin theme addon on";
            example = lib.literalExpression ''[ "default" "default-release" "work" ]'';
          };
```

---

Why not? Is it not the case that everyone starts out with a profile named "default" unless they add a new one?

---

I don't think so, for example, I use my own name for my default profile.

Doing a quick seach through github search, you can see many people don't use `default` as a profile:

https://github.com/ksevelyar/idempotent-desktop/blob/04d0508ffa28dea3e57ab23e54414f722d64551f/packages/firefox.nix#L11

https://github.com/skogsbrus/os/blob/ba9c523d8b3b086b54a1a644b1df779dbb39e1c2/home/firefox.nix#L16

https://github.com/icyphox/dotfiles/blob/76fda9709b55a6e0a6ec775c307d7bec52fb28db/programs/firefox.nix#L9

You can see more in here:

https://github.com/search?q=profiles+AND+programs+AND+firefox+language%3ANix&type=code

---

There are plenty of people that use `default` as a profile. But this should be up to the user to configure anyways, since this is not a default that fits everyone

---

I meant that if no profiles were configured, then `default` would be the name of the profile.

In any case, it occurred to me that most people using this would explicitly configure their profile in their config anyways, so I went ahead and added your suggestion.